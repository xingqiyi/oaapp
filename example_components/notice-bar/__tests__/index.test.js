import React from 'react';
import { render, shallow } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import NoticeBar from '../index';

describe('NoticeBar', () => {
  it('renders correctly', () => {
    const wrapper = render(<NoticeBar>foo</NoticeBar>);
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  describe('onClick', () => {
    let handleClick;
    let wrapper;

    beforeEach(() => {
      handleClick = jest.fn();
      wrapper = shallow(
        <NoticeBar mode="closable" onClick={handleClick}>
          Notice: The arrival time of incomes and
        </NoticeBar>,
      );
      wrapper.find('.am-notice-bar-operation').simulate('click');
    });

    it('fires onClick event', () => {
      expect(handleClick).toBeCalledWith();
    });

    it('change state', () => {
      expect(wrapper.state('show')).toBe(false);
    });
  });
});
