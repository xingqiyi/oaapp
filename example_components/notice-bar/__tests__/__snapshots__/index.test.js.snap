// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoticeBar renders correctly 1`] = `
<div
  class="am-notice-bar"
  role="alert"
>
  <div
    aria-hidden="true"
    class="am-notice-bar-icon"
  >
     
    <svg
      class="am-icon am-icon-trips am-icon-xxs"
    >
      <use
        xlink:href="#trips"
      />
    </svg>
     
  </div>
  <div
    class="am-notice-bar-content"
  >
    <div
      class="am-notice-bar-marquee-wrap "
      role="marquee"
      style="overflow:hidden;"
    >
      <div
        class="am-notice-bar-marquee"
        style="position:relative;right:0;white-space:nowrap;display:inline-block;"
      >
        foo 
      </div>
    </div>
  </div>
</div>
`;
